### PR TITLE
update imports and types of module "xml-zero-lexer"

### PR DESCRIPTION
**Pull request by 943297456**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,9 @@ resources/IndexData/index/*.cfe
 resources/IndexData/index/*.cfs
 resources/IndexData/index/*.si
 resources/IndexData/index/segments*
+
+.vs/*
```
Comment (Eskibear) [inline .gitignore:19]: what for?

**Comment by Eskibear** [comment]
Suggest that you directly help to update and publish `@types/xml-zero-lexer`. So we can import it in TS code without the duplicate definition.

File: .gitignore (hunk 1 above)
Comment (943297456) [reply .gitignore:19]: some files are generated by Visual Studio IDE

**Comment by Eskibear** [comment]
I'm ok to merge this PR with below asks:
* can you remove `.vscode` from `.gitignore`, as we do want to include vscode-related files, including whatever will be in that folder in the future. 
* do you have the interest to update and publish `@types/xml-zero-lexe`?

File: .gitignore (hunk 1 above)
Comment (Eskibear) [reply .gitignore:19]: ok jusjt keep it, but please remove the `.vscode` part.

**Comment by 943297456** [comment]
> Suggest that you directly help to update and publish `@types/xml-zero-lexer`. So we can import it in TS code without the duplicate definition.


That's a good idea.

I'll take a moment to learn more about this module and provide its definition.

**Pull request merged**